java/code/src/com/suse/manager/virtualization/VirtManager.java

```diff
@@ -80,6 +80,29 @@
         return saltService.callSync(call, minionId);
     }
 
+    /**
+     * Query virtual storage pool definition
+     *
+     * @param minionId the host minion ID
+     * @param poolName the domain name to look for
+     * @return the XML definition or an empty Optional
+     */
+    public static Optional<PoolDefinition> getPoolDefinition(String minionId, String poolName) {
+        Map<String, JsonObject> infos = getPools(minionId);
+
+        Map<String, Object> args = new LinkedHashMap<>();
+        args.put("name", poolName);
+        LocalCall<String> call =
+                new LocalCall<>("virt.pool_get_xml", Optional.empty(), Optional.of(args), new TypeToken<String>() { });
+
+        Optional<String> result = saltService.callSync(call, minionId);
+        return result.filter(s -> !s.startsWith("ERROR")).map(xml -> {
+            PoolDefinition pool = PoolDefinition.parse(xml);
+            pool.setAutostart(infos.get(poolName).get("autostart").getAsInt() == 1);
```

Comment:
I think this can throw a `NullPointerException` if for some reason the parsing fails and `pool` is null.

Reply:
handled